app/src/main/java/com/dankideacentral/dic/MainActivity.java

```diff
@@ -72,9 +80,20 @@ public void failure(TwitterException exception) {
                 Log.d("TwitterKit", "Login with Twitter failure", exception);
             }
         });
-        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
-                .findFragmentById(R.id.map);
-        mapFragment.getMapAsync(this);
+        toggleButton = (Button) findViewById(R.id.toggle);
+        toggleButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                currentFragment = currentFragment instanceof ItemFragment ?
+                        mapFragment :
+                        new ItemFragment();
+                FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+                transaction.replace(R.id.frame_container, currentFragment);
+                transaction.addToBackStack(null);
```

Comment:
when we implement the back button from the itemFragment to the map we should add the map view (fragment) to the back stack. null is good for now.